crates/uv-workspace/src/settings.rs

```diff
@@ -255,6 +267,10 @@ impl Options {
             upgrade_package: top_level_upgrade_package,
             reinstall: top_level_reinstall,
             reinstall_package: top_level_reinstall_package,
+            no_build: _,
+            no_build_package: _,
+            no_binary: _,
+            no_binary_package: _,
```

Comment:
So... this means that the `no-build`, `no-build-package`, `no-binary`, and `no-binary-package` settings on `tool.uv` are ignored on the `pip` API (which has its own `--no-binary`, `--only-binary`, etc.).

I guess this makes sense? I'm not sure what else to do, because the APIs are totally different (the types don't line up).

Reply:
That does seem weird. Do they need to be? Isn't our underlying type the same and we support combination?

Reply:
They are certainly not the same type here. Perhaps if I get rid of this method and inline it into `PipSettings` I can... find a way to combine them. I'm not sure. I will try... I don't know if the semantics are clear, lets see if I run into problems.

Reply:
Yeah we can't combine them here. We would need to move this merging logic downstream. It is probably doable.

Reply:
As far as I know the semantics should be the same it's just a difference in the user-facing types. I could look at it after too if you want.

Reply:
Yeah it's just a bit less clear because the variables don't line up, so we can't (e.g.) `no_binary.or(other.no_binary)`. So I just need to validate the combination logic.

Reply:
Yeah, but they _do_ line up once they're cast into `BuildOptions` (and its children) right? Might be easier to just do that early or pass everything through until we get to the point where we construct that and combine them there.

Reply:
Got this to work as expected, yes.

Reply:
`BuildOptions` would make things more slightly difficult because we actually read _more_ `--no-binary` arguments from `requirements.txt` files later on.

Reply:
(We do convert eagerly to `BuildOptions` for the non-pip APIs. Let me see if we can do the same for the pip APIs.)

Reply:
Done...